modules/centrifuge/main.nf

```diff
@@ -0,0 +1,38 @@
+process CENTRIFUGE {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::centrifuge=1.0.4_beta" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/centrifuge:1.0.4_beta--h9a82719_6' :
+        'quay.io/biocontainers/centrifuge:1.0.4_beta--h9a82719_6' }"
+
+    input:
+    tuple val(meta), path(reads)
+    path db
+
+    output:
+    tuple val(meta), path('*report.txt')  , emit: report
+    path "versions.yml"                   , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def paired = meta.single_end ? "-U ${reads}" :  "-1 ${reads[0]} -2 ${reads[1]}"
+    """
+    tar -xf $db
+    centrifuge \\
+        -x ${db.toString().replace(".tar.gz","")} \\
+        -p $task.cpus \\
+        $paired \\
+        --report-file ${prefix}.centrifuge.report.txt \\
+        $args
+    cat <<-END_VERSIONS > versions.yml
```

Comment:
```suggestion

    cat <<-END_VERSIONS > versions.yml
```

Reply:
I was thinking if those should be optional? Taking into account how centrifuge is implemented in mag.
 ```
 --un <path>           write unpaired reads that didn't align to <path>
 --al <path>           write unpaired reads that aligned at least once to <path>
 --un-conc <path>      write pairs that didn't align concordantly to <path>
 --al-conc <path>      write pairs that aligned concordantly at least once to <path>
 (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.
 --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)
```

Reply:
What about `-al`? Also I see there is the `--met-file` option... I'm not entirely sure what that means here but it might be useful if it containts logging information?

Reply:
I attach an example of how `--met-file` option looks like:
[met.txt](https://github.com/nf-core/modules/files/8243538/met.txt)

Reply:
Ok `--met-file` looks like developer debugging things, I think we can skip that :)

Reply:
Regarding the `-al` and `-un`, would that name convention work? Or would you prefer skipping `unmapped`, `mapped` in the filename?

```
def unaligned = ''
def aligned = ''
if (meta.single_end) {
     unaligned = save_unaligned ? "--un-gz ${prefix}.unmapped.fastq.gz" : ''
     aligned = save_aligned ? "--al-gz ${prefix}.mapped.fastq.gz" : ''
} else {
      unaligned = save_unaligned ? "--un-conc-gz ${prefix}.unmapped.fastq.gz" : ''
      aligned = save_aligned ? "--al-conc-gz ${prefix}.mapped.fastq.gz" : ''
}
```

Reply:
In this case we would indeed have to specify. I think it's is fine in this csae :+1: